Covid/Map/SpreadViewController.swift

```diff
@@ -55,65 +56,96 @@ final class RegionInfo: NSObject, Codable {
 struct LocationJSON: Codable {
     let lat: Double
     let lon: Double
+    
+    var coordinate: CLLocationCoordinate2D {
+        CLLocationCoordinate2D(latitude: lat, longitude: lon)
+    }
 }
 
 final class SpreadViewController: UIViewController {
+    enum ViewType: Int {
+        case map
+        case list
+    }
+    
     @IBOutlet private var mapContainerView: UIView!
     @IBOutlet private var listContainerView: UIView!
-
+    
+    @IBOutlet private weak var segmentedControl: UISegmentedControl!
     private var mapViewController: SpreadMapViewController?
     private var listViewController: SpreadListTableViewController?
-
+    
     private var data = [RegionInfo]()
-
+    
     override func viewDidLoad() {
-    super.viewDidLoad()
-
+        super.viewDidLoad()
+        
         mapViewController = children.first { $0 is SpreadMapViewController } as? SpreadMapViewController
         listViewController = children.first { $0 is SpreadListTableViewController } as? SpreadListTableViewController
+        listViewController?.onRegionSelect = { [weak self] regionInfo in
+            self?.selectRegion(with: regionInfo)
+        }
     }
-
+    
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
-
+        
         loadData()
     }
-
+    
+    // MARK: region select
+    private func selectRegion(with regionInfo: RegionInfo) {
+        switchToViewType(.map) { [weak self] in
+            self?.mapViewController?.selectAnnotation(with: regionInfo)
+        }
+    }
+    
+    private func switchToViewType(_ viewType: ViewType, completion: (() -> Void)? = nil) {
+        UIView.animate(withDuration: 0.3,
+                       animations: { [weak self] in
+                        self?.switchToViewType(viewType)
+            }, completion: { _ in
+                completion?()
+        })
+    }
+    
+    private func switchToViewType(_ viewType: ViewType) {
```

Comment:
akoze tu je to asi jedno, ale metoda s rovnakym nazvom, len ci ma alebo nema completion sposobi ze to robi animovane, to zavana problemom 😅

Reply:
@anonymcek haha, hej, tiez som sa nad tym nachvilenku pozastavil o polnoci, no bol som lenivy to pomenovat inak, napravim ;)